tests/test_condense.py

```diff
@@ -0,0 +1,356 @@
+    if dict_functional_tensors["inputs"] is None:
+        return outputs
+    else:
+        return inputs, outputs
+    # return x
+
+
+def get_model(layer_type, layer_params, input_shape, k_coef_lip):
+    if layer_type == tModel:
+        return uft.get_functional_model(
+            tModel,
+            layer_params["dict_tensors"],
+            layer_params["functional_input_output_tensors"],
+        )
+    else:
+        return uft.generate_k_lip_model(
+            layer_type, layer_params, input_shape=input_shape, k=k_coef_lip
+        )
+
+
+@pytest.mark.skipif(
+    hasattr(SpectralConv2dTranspose, "unavailable_class"),
+    reason="SpectralConv2dTranspose not available",
+)
+@pytest.mark.parametrize(
+    "layer_type, layer_params, k_coef_lip, input_shape",
+    [
+        (Sequential, {"layers": sequential_layers((3, 8, 8))}, 5.0, (3, 8, 8)),
+        (
+            tModel,
+            {
+                "dict_tensors": get_functional_tensors((3, 8, 8)),
+                "functional_input_output_tensors": functional_input_output_tensors,
+            },
+            5.0,
+            (3, 8, 8),
+        ),
+    ],
+)
+def test_model(layer_type, layer_params, k_coef_lip, input_shape):
+    batch_size = 250
+    epochs = 1
+    steps_per_epoch = 125
+    k_lip_data = 2.0
+
+    # clear session to avoid side effects from previous train
+    uft.init_session()  # K.clear_session()
+    np.random.seed(42)
+    input_shape = uft.to_framework_channel(input_shape)
+
+    model = get_model(layer_type, layer_params, input_shape, k_coef_lip)
+
+    # create the keras model, defin opt, and compile it
+    optimizer = uft.get_instance_framework(
+        uft.Adam, inst_params={"lr": 0.001, "model": model}
+    )
+
+    loss_fn, optimizer, metrics = uft.compile_model(
+        model,
+        optimizer=optimizer,
+        loss=uft.MeanSquaredError(),
+        metrics=[uft.metric_mse()],
+    )
+    # create the synthetic data generator
+    output_shape = uft.compute_output_shape(input_shape, model)
+    # output_shape = model.uft.compute_output_shape((batch_size,) + input_shape)[1:]
+    kernel = build_kernel(input_shape, output_shape, k_lip_data)
+    # define logging features
+    logdir = os.path.join(
+        "logs", uft.LIP_LAYERS, "condense_test"
+    )  # , datetime.now().strftime("%Y_%m_%d-%H_%M_%S")))
+    os.makedirs(logdir, exist_ok=True)
+    callback_list = []  # callbacks.TensorBoard(logdir)]
+    # train model
+
+    traind_ds = linear_generator(batch_size, input_shape, kernel)
+    uft.train(
+        traind_ds,
+        model,
+        loss_fn,
+        optimizer,
+        epochs,
+        batch_size,
+        steps_per_epoch=steps_per_epoch,
+        callbacks=callback_list,
+    )
+    # model.__getattribute__(FIT)(
+    #     linear_generator(batch_size, input_shape, kernel),
+    #     steps_per_epoch=steps_per_epoch,
+    #     epochs=epochs,
+    #     verbose=0,
+    #     callbacks=callback_list,
+    # )
```

Comment:
We can remove the code from TensorFlow/Keras in comments at different places in this file

Reply:
the commit 26fd0e30791e956a135fce6f1ea71d3c02ccbd05 solved this issue